server/diagnostics.jai

```diff
@@ -25,7 +25,7 @@ run_diagnostics :: () {
     command: [..] string;
     defer array_free(command);
 
-    array_add(*command, "jai");
+    array_add(*command, server.jai_path);
```

Comment:
#51 for this actual change in behaviour, but ofc that uses `server.args` so I changed it here too.

Reply:
#51 got more complicated, so this isn't the right change any more, reverting, whichever goes in first - this or #51 - the other one will need a bit of reworking.